src/main/java/org/apereo/openlrs/controllers/caliper/CaliperController.java

```diff
@@ -15,44 +15,134 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apereo.openlrs.exceptions.NotFoundException;
+import org.apereo.openlrs.exceptions.caliper.InvalidCaliperFormatException;
+import org.apereo.openlrs.exceptions.xapi.InvalidXAPIRequestException;
 import org.apereo.openlrs.model.caliper.CaliperEvent;
+import org.apereo.openlrs.model.xapi.About;
 import org.apereo.openlrs.services.caliper.CaliperService;
+import org.apereo.openlrs.utils.StatementUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.*;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
 
 /**
  * @author ggilbert
- *
+ * @author Lance E Sloan (lsloan at umich dot edu)
  */
 @RestController
 @RequestMapping("/caliper")
 public class CaliperController {
-  
-  private final Logger logger = LoggerFactory.getLogger(CaliperController.class);
-  @Autowired private ObjectMapper objectMapper;
-  @Autowired private CaliperService caliperService;
-
-  @RequestMapping(value = {"", "/"}, method = RequestMethod.POST)
-  public String event(@RequestBody String json) throws JsonProcessingException, IOException {
-    logger.debug(json);
-    
-    JsonNode jsonNode = objectMapper.readTree(json);
-    logger.debug("Type: "+jsonNode.findValue("@type").textValue());
-    CaliperEvent caliperEvent = new CaliperEvent(jsonNode);
-    caliperService.post(null, caliperEvent);
-    return caliperEvent.getKey();
-  }
-
-}
+    private final Logger logger = LoggerFactory.getLogger(CaliperController.class);
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private CaliperService caliperService;
+    @Value("${caliper.version}")
+    private String version;
+
+
+    /**
+     * @param statementId      ID of the statement to find
+     * @param allRequestParams other search parameters
+     * @return JSON node of requested Caliper event
+     */
+    @RequestMapping(value = {"", "/"}, method = RequestMethod.GET,
+            produces = "application/json;charset=utf-8", params = "statementId")
+    public JsonNode getByIdHandler(
+            @RequestParam(value = "statementId", required = true) String statementId,
+            @RequestParam Map<String, String> allRequestParams) {
```

Comment:
Yes.  I patterned my method after `StatementController.getStatement()`.  It had the `allRequestParams` argument, but it didn't really do anything with it either.  I thought maybe the argument was required by the framework, but I admit I didn't test the application without that argument.  I'll give it a try and if it works, I'll commit the change.